plugins/ccgx/fu-ccgx-pure-hid-device.c

```diff
@@ -0,0 +1,514 @@
+/*
+ * Copyright (C) 2023 Framework Computer Inc
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <string.h>
+
+// #include "fu-ccgx-common.h"
+#include "fu-ccgx-firmware.h"
+#include "fu-ccgx-hpi-common.h"
+#include "fu-ccgx-pure-hid-device.h"
+#include "fu-ccgx-struct.h"
+
+#define ROW_SIZE	  128
+#define FW1_START	  0x0030
+#define FW2_START	  0x0200
+#define FW1_METADATA	  0x03FF
+#define FW2_METADATA	  0x03FE
+
+/**
+ * FU_CCGX_PURE_HID_DEVICE_IS_IN_RESTART:
+ *
+ * Device is in restart and should not be closed manually.
+ *
+ * Since: 1.9.2
```

Comment:
this should bump up now

Reply:
Ohh, I think @d4s has a better branch than this one now.

Reply:
ohh... not yet. Just did some cleanup and minor fixes.
Let me fix the rest of TODO/FIXME tomorrow and test updates.